### ISSUE TITLE
A proposal to move the content in ethereum/wiki to a Wikipedia-style wiki site

### ISSUE THREAD

**Issue by jamesray1** [ethereum/wiki#589]
> Github wikis have the limitation that anyone can edit them while it is difficult to moderate them (it is not possible to watch a page to get updates for changes), which is a double-edged sword, in that it is great for non-censorship which is in the spirit of the [Ethereum philosophy](https://github.com/ethereum/wiki/wiki/White-Paper#philosophy), however it is also prone to graffiti, bias, etc. Because of this limitation of lack of ease for moderation, we should consider moving the content in this wiki to a wiki site, e.g. one that is hosted by [Media Wiki](https://www.mediawiki.org/wiki/MediaWiki), adding a deprecation notice to this wiki and refer to the proposed new Media Wiki hosted site. This would have the benefits of anyone still being able to edit the wiki (they do not even need to create an account, although there are advantages to doing so), as well as being able to more easily moderate content.
— https://github.com/ethereum/wiki/wiki#a-proposal-to-move-the-content-in-this-wiki-to-a-wikipedia-style-wiki-site

Also posted in https://ethereum-magicians.org/t/a-proposal-to-move-the-content-in-ethereum-wiki-to-a-wikipedia-style-wiki-site/265 and https://github.com/ethereum/EIPs/issues/1053.

@pirapira

**Comment by jamesray1** [comment]
Related: https://github.com/ethereum/wiki/issues/591.

**Comment by jamesray1** [comment]
See https://www.quora.com/Whats-a-good-host-for-a-wiki-site-that-has-watch-ability-for-pages-with-email-notifications-for-changes-and-compatibility-with-GitHub-Flavoured-Markdown for a potential solution.

**Comment by jamesray1** [comment]
Just clarifying that I don’t want to work on moving the wiki to a Mediawiki hosted site because I need to focus on sharding development and get a grant and funds (so far my work, including editing this wiki and other docs, has gone unremunerated). I already frequently update the wiki.

**Comment by fubuloubu** [comment]
We are looking at gitbook for SecurEth, it is definitely pretty interesting so far. I like that it can be backed by GitHub so you can have tighter control over how updates get made.

**Comment by jamesray1** [comment]
I already looked into that plus other alternatives, but it doesn't allow anyone to edit like Mediawiki sites, yet also have better moderation features like watching pages.

Also, if moving content to a Mediawiki site, all the cross-links to elsewhere within the wiki would need to be updated.

**Comment by virgil** [comment]
I have a beta working at: http://ethereum.wiki

**Comment by jamesray1** [comment]
Great! Would be good if there is a way to watch a page and get notifications for changes, plus other features for moderators. See https://www.quora.com/Whats-a-good-host-for-a-wiki-site-that-has-watch-ability-for-pages-with-email-notifications-for-changes-and-compatibility-with-GitHub-Flavoured-Markdown for a potential solution for that with a MediaWiki-hosted site.

**Comment by virgil** [comment]
Yeah I tried MediaWiki---I wasn't able to get it as pretty as I desired.  Also the editing was kinda meh for markdown.

I think as-is I'm pretty committed to WikiJS.  There is a new version coming out (WikiJS 2.0) where we'll be able to get feature requests like that.

Note that you can get history pages on WikiJS, for example: https://en.ethereum.wiki/hist/home

**Comment by jamesray1** [comment]
Fair enough, I did remember WikiJS mentioning that update, let's wait until then.

**Comment by jamesray1** [comment]
I do like the sidebar of this github wiki/wiki, while that's not on the current site. It makes it easier to navigate and categorise things. It's good that the search bar is displayed top centre on the site.

**Comment by jamesray1** [comment]
I guess that the disadvantages of having a sidebar is that there is less space for content, and it could become cluttered over time as more information is developed about the Ethereum ecosystem.

**Comment by w7788129** [comment]
Go

**Comment by jamesray1** [comment]
I just looked at https://en.ethereum.wiki/hist/home. One paragraph appears as one line; and there is no soft wrap option, let alone by default. This doesn't occur if you edit a page, just in comparing revisions.

Pages also appear to be out of sync, e.g. https://en.ethereum.wiki/edit/famous-bugs and https://github.com/ethereum/wiki/wiki/Bugs.

Issues so far:
- [ ] No soft wrap in revision history
- [ ] while history for a page is accesible by modifying the link, e.g. https://en.ethereum.wiki/faqs -> https://en.ethereum.wiki/hist/faqs, this is not a good UX; there should be a button to access history.
- [ ] no sidebar
- [ ] not able to search for content in pages
- [ ] pages appear to be out of sync, e.g. https://en.ethereum.wiki/edit/famous-bugs and https://github.com/ethereum/wiki/wiki/Bugs. https://github.com/ethereum/wiki/wiki/Decentralized-apps-(dapps) is missing (on the WikiJS site). https://en.ethereum.wiki/fundamentals/getting-ether exists, but it was confusing to navigate to https://en.ethereum.wiki/fundamentals (click all pages in the sidebar, then click the arrow in fundamentals), which should probably not be a page. If you don't click the arrow but just click anywhere else on that Fundamentals line, it will take you to a list of sub pages in the sidebar.

I would probably consider any one of the above issues to be a deal breaker to using the wikiJS site over the GIthub wiki, especially without wikiJS 2.0.

Also, there is no ability yet to watch a page and other features for moderators, but this can't be considered an issue as Github wikis do not provide this feature yet either, nor have they committed to doing so, while it is planned for WikiJS 2.0.

**Comment by virgil** [comment]
1. You're right that the lack of soft-wrap sucks.  I suppose we can do a pull request for it---that's an easy fix.
2. The UX for history comparison does indeed suck.  It's an undocumented and never finished feature.
3. Wait, there is a sidebar, it's just on the left.  See: https://en.ethereum.wiki/home
4. You can.  Maybe it's just complete words?  E.g., search for "japanese" in the top bar and you'll see the the "Home" link.
5. The synced github repos are:
* https://github.com/ethresearch/en-ethereum-wiki
* https://github.com/ethresearch/zh-ethereum-wiki
* https://github.com/ethresearch/ja-ethereum-wiki

I tried a few other wikis, my second choice was DokuWiki.  But WikiJS was the best I found and I agree completely that we will upgrade to WikiJS 2.0 once it exists.

**Comment by jamesray1** [comment]
> Wait, there is a sidebar, it's just on the left.

Right, I guess it needs better categorisation into folders like on https://github.com/ethereum/wiki/wiki, which I prefer as it targets beginners with basics at the top, followed by R&D (which is arguably more important for scalability and thus overall use of Ethereum, than client development, dapp development, infrastructure, etc.) The sidebar on the Github wik categorises key pages so that you don't have to go to "all pages" to find frequently used pages.

> You can. Maybe it's just complete words? E.g., search for "japanese" in the top bar and you'll see the the "Home" link.

Hmmm, maybe the site has been crawled recently as when I searched for e.g. "Plasma" before, it didn't come up in the sharding FAQs, but now it does.

> The synced github repos are:

I see, so changes to https://github.com/ethereum/wiki/wiki are not synced with https://en.ethereum.wiki and vice versa. That could be problematic in between the transition to deactivating the former and only using the latter.

**Comment by virgil** [comment]
I'm wholly down for moving these into the official Ethereum repo. I often use the ethresearch GitHub as a sandbox before things are ready for primetime. If you'd prefer to move this over to the Ethereum GitHub now I'm all for it.

**Comment by jamesray1** [comment]
It'll need to be done eventually, otherwise discrepancies between the beta site and the wiki/wiki will continue to widen. As long as syncing doesn't cause anything untoward then I don't see why not, and I guess it could be reverted if needed.

**Comment by virgil** [comment]
We probably don't want make github wiki an official frontend because Github Markdown is slightly different from regular markdown (which WikiJS uses).  However, if the ethereum github clones the wiki repos on ethresearch I will set the backends to the repos on the ethereum github instead of ethresearch.

**Comment by jamesray1** [comment]
I think the problem with that is if the Ethereum Github wiki clones the ethresearch repo, which is syncing with ethereum.wiki, some data will be lost, e.g. missing pages on ethereum.wiki like https://github.com/ethereum/wiki/wiki/Decentralized-apps-(dapps); and edits made that aren't showing on ethresearch, e.g. this has edits made 4 days ago: https://github.com/ethereum/wiki/wiki/Decentralized-apps-(dapps)/_history, but this was last edited 18 days ago: https://github.com/ethresearch/en-ethereum-wiki/blob/master/decentralized-apps-(dapps).md. I checked that the Decentralized-apps-(dapps) page is not missing on the ethresearch repo: https://github.com/ethresearch/en-ethereum-wiki/blob/master/decentralized-apps-(dapps).md.

**Comment by virgil** [comment]
I will volunteer to personally see to it that house edits are included in the new wiki.

**Comment by jamesray1** [comment]
OK. I'd like to be able to categorise the "all pages" better before moving to the new wiki. E.g. like on the sidebar here https://github.com/ethereum/wiki/wiki with a categories for R&D; clients, tools, dapp browsers; APIs; languages; etc; infrastructure; dev technologies; Ethereum technologies, etc.

**Comment by virgil** [comment]
Sure.  Will include any of the sidebars you'd like.

**Comment by virgil** [comment]
I've made the sidebar on https://en.ethereum.wiki/home via tables.  It's not pretty, but it's not terrible.  We can probably do better.

I have re-organized a few bits putting things into subfolders, etc.  I'm not wedded to the categories I chose.  If you want a different structure I have no complaints.

**Comment by jamesray1** [comment]
Thanks, in one sense it's actually easier to navigate to those pages as you don't have to scroll down the page. A disadvantage is that it's only on the home page, not every page, but at least you can get to the home page easily.

**Comment by hivemall** [comment]
newb diving into eth wiki, i would soo appreciate default wikipedia style, the tools are opensource; dont reinvent the wheel. @jamesray1 excellent suggestion

**Comment by jamesray1** [comment]
Moving to a Wikipedia style would involve either making it compatible with GitHub Flavoured Markdown, or updating the wiki to conform to Wikipedia syntax. The former should be easier, as mentioned above, see https://www.quora.com/Whats-a-good-host-for-a-wiki-site-that-has-watch-ability-for-pages-with-email-notifications-for-changes-and-compatibility-with-GitHub-Flavoured-Markdown for a solution. It would be less effort to wait for wiki-JS 2.0.

**Comment by nhandl3** [comment]
Sorry I don't have much experience with MediaWiki yet but I have a question. You @jamesray1 said that it's hard to moderate change with Github's Wiki. Why don't we use Github's repo itself to manage changes? With repo it can fully version control, making issue/PR for change request & review. There are some projects documenting this way (Github themselves) and pretty good in control.

For example `atom` project, this final page https://flight-manual.atom.io/getting-started/sections/why-atom/  and here is its repo history https://github.com/atom/flight-manual.atom.io/commits/master/content/getting-started/sections/why-atom.md

**Comment by jamesray1** [comment]
Because people can make changes to the wiki directly. While spam edits and the like occasionally happen, IMO the wiki is still better as it saves needing a maintainer to review trivial edits. With no maintainer allocated to the task and dozens of PRs being left open and unattended, I decided to try to point people towards being more proactive by editing themselves, filing an issue with relevant parties e.g. for Mist, web3, etc. A wiki is more decentralized, being freely maintainable by the community, rather than being controlled by the Ethereum Foundation.

If you are interested in being a maintainer you could try talking to someone in the Ethereum Foundation such as @Souptacular.

**Comment by jamesray1** [comment]
Someone has removed the sidebar and changed the footer to an ethereum address. :(

**Comment by virgil** [comment]
Here we go, https://eth.wiki .

**Comment by revolution001** [comment]
jam ray u play me???

**Comment by revolution001** [comment]
jame ray out  site

**Comment by revolution001** [comment]
i can stuck overall u want fight me
